Cut(eos_designs)!: Remove deprecated keys for terminattr_smashexcludes, terminattr_ingestexclude, terminattr_disable_aaa, terminattr_ingestgrpcurl_port, cvp_ingestauth_key, cvp_instance_ips and cvp_token_file

Vibhu-gslab opened the pull request:
## Change Summary

 Remove deprecated keys for terminattr_smashexcludes, terminattr_ingestexclude, terminattr_disable_aaa, terminattr_ingestgrpcurl_port, cvp_ingestauth_key, cvp_instance_ips and cvp_token_file

## Related Issue(s)

Part of https://github.com/aristanetworks/avd-internal/issues/300

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Removing the deprecated keys

## Checklist

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

Vibhu-gslab committed:
Cut(eos_designs): Remove deprecated keys for terminattr_smashexcludes, terminattr_ingestexclude and terminattr_disable_aaa

MaheshGSLAB commented on python-avd/pyavd/_eos_designs/structured_config/base/daemon_terminattr.py:
```diff
@@ -138,9 +138,4 @@ def _legacy_daemon_terminattr(self: AvdStructuredConfigBaseProtocol) -> None:
-            ingestexclude=self.inputs.terminattr_ingestexclude,
-            disable_aaa=self.inputs.terminattr_disable_aaa,
-        )
+        self.structured_config.daemon_terminattr._update(cvvrf=self.inputs.mgmt_interface_vrf)
```
```suggestion
        self.structured_config.daemon_terminattr.cvvrf = self.inputs.mgmt_interface_vrf
```

Vibhu-gslab committed:
Update python-avd/pyavd/_eos_designs/structured_config/base/daemon_terminattr.py

Co-authored-by: Mahesh Kumar <122076792+MaheshGSLAB@users.noreply.github.com>

MaheshGSLAB commented on python-avd/pyavd/_eos_designs/structured_config/base/daemon_terminattr.py:
```diff
@@ -138,9 +138,4 @@ def _legacy_daemon_terminattr(self: AvdStructuredConfigBaseProtocol) -> None:
-            ingestexclude=self.inputs.terminattr_ingestexclude,
-            disable_aaa=self.inputs.terminattr_disable_aaa,
-        )
+        self.structured_config.daemon_terminattr.cvvrf = self.inputs.mgmt_interface_vrf
```
cvp_instance_ips is also deprecated so include it this PR which will remove the whole method

MaheshGSLAB replied:
cvp_ingestauth_key, terminattr_ingestgrpcurl_port and cvp_token_file too

Vibhu-gslab replied:
Done!

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
including cvp_ingestauth_key, cvp_instance_ips and cvp_token_file

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

MaheshGSLAB commented on ansible_collections/arista/avd/roles/eos_designs/docs/tables/cloudvision-settings.md:
```diff
@@ -129,51 +128,10 @@
     # This key is deprecated.
     # Support will be removed in AVD version 6.0.0.
     # Use `cv_settings` instead.
     terminattr_ingestgrpcurl_port: <int; default=9910>
```
I tink we missed to remove the key terminattr_ingestgrpcurl_port. Please remove that one as well

Vibhu-gslab replied:
Done!

Vibhu-gslab committed:
including terminattr_ingestgrpcurl_port as well

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Shivani-gslab reviewed (commented):
LGTM

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

ClausHolbechArista commented on docs/porting-guides/6.x.x.md:
```diff
@@ -21,6 +21,9 @@ Users of `eos_designs` need to consider the changes in `eos_cli_config_gen` when
 
+## Changes to role `arista.avd.eos_designs`
+
+=======
```
```suggestion
```

ClausHolbechArista commented on docs/porting-guides/6.x.x.md:
```diff
@@ -59,7 +62,14 @@ The following data model keys have been removed from `eos_designs` in AVD 6.0.0.
+| `terminattr_ingestexclude` | `cv_settings.terminattr.ingestexclude` |
+| `terminattr_smashexcludes` | `cv_settings.terminattr.smashexcludes` |
+| `terminattr_ingestgrpcurl_port` | `cv_settings` |
 
```
Let's add a section on how to migrate from the many cv related vars into cv_settings.

JulioPDX replied:
Added an initial example for the migration. @Vibhu-gslab feel free to modify/use as inspiration.

Vibhu-gslab replied:
Thanks @JulioPDX , I just added 2 more attributes. Rest all looks good to me :)

JulioPDX committed:
adding migration section

gmuloc committed:
Update docs/porting-guides/6.x.x.md

Co-authored-by: Claus Holbech <holbech@arista.com>

gmuloc commented on docs/porting-guides/6.x.x.md:
```diff
@@ -59,7 +59,65 @@ The following data model keys have been removed from `eos_designs` in AVD 6.0.0.
+| `terminattr_disable_aaa` | `cv_settings.terminattr.disable_aaa` |
+| `terminattr_ingestexclude` | `cv_settings.terminattr.ingestexclude` |
+| `terminattr_smashexcludes` | `cv_settings.terminattr.smashexcludes` |
+| `terminattr_ingestgrpcurl_port` | `cv_settings` |
```
```suggestion
| `terminattr_ingestgrpcurl_port` | `cv_settings` |
| `terminattr_smashexcludes` | `cv_settings.terminattr.smashexcludes` |
```

gmuloc committed:
Update docs/porting-guides/6.x.x.md

gmuloc commented on python-avd/pyavd/_eos_designs/structured_config/base/daemon_terminattr.py:
```diff
@@ -28,10 +28,6 @@ def daemon_terminattr(self: AvdStructuredConfigBaseProtocol) -> None:
 
         The schema will enforce that we only use either new or old models.
         """
-        if not self.inputs.cv_settings:
-            self._legacy_daemon_terminattr()
-            return
```
why do you remove the early return? You should keep

```
        if not (cv_settings := self.inputs.cv_settings):
            return
```

and you can then remove line 35

Vibhu-gslab replied:
Updated with Walrus :)

Vibhu-gslab committed:
adding back cv_settings check

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
using Walrus

Vibhu-gslab committed:
Update 6.x.x.md

alexeygorbunov commented on python-avd/pyavd/_eos_designs/schema/schema_fragments/terminattr_ingestgrpcurl_port.schema.yml:
```diff
@@ -18,5 +18,6 @@ keys:
       The port number is always 443 when using CloudVision as a Service, so this value is ignored.
     deprecation:
       warning: true
+      removed: true
```
Why description is not removed for this key?

Vibhu-gslab replied:
Missed it. 
Removed now :)

alexeygorbunov commented:
Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):

```yaml
keys:
  inband_ztp_bootstrap_file:
    type: str
    description: |-
      Bootstrap URL configured in DHCP to use for inband ZTP.
      If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
          `https://{cvp_instance_ips[0]}/ztp/bootstrap`
      Otherwise no value will be configured.
```

Vibhu-gslab committed:
updating schemas

Vibhu-gslab commented:
> Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> 
> ```yaml
> keys:
>   inband_ztp_bootstrap_file:
>     type: str
>     description: |-
>       Bootstrap URL configured in DHCP to use for inband ZTP.
>       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
>           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
>       Otherwise no value will be configured.
> ```

I have replaced it but not sure about the change, please have a look.

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

alexeygorbunov commented:
> > Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> > ```yaml
> > keys:
> >   inband_ztp_bootstrap_file:
> >     type: str
> >     description: |-
> >       Bootstrap URL configured in DHCP to use for inband ZTP.
> >       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
> >           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
> >       Otherwise no value will be configured.
> > ```
> 
> I have replaced it but not sure about the change, please have a look.

It'll use either CVaaS (if set) or the first server of the first onprem cluster. Maybe something like this then?

```diff
    description: |-
      Bootstrap URL configured in DHCP to use for inband ZTP.
      If not set then the bootstrap value will be set to either `https://www.{CV_REGION_TO_SERVER_MAP[next(iter(self.inputs.cv_settings.cvaas.clusters)).region]}/ztp/bootstrap` or
      `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
      Otherwise no value will be configured.
```

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

alexeygorbunov committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

gmuloc commented:
> > > Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> > > ```yaml
> > > keys:
> > >   inband_ztp_bootstrap_file:
> > >     type: str
> > >     description: |-
> > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > >       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
> > >           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
> > >       Otherwise no value will be configured.
> > > ```
> > 
> > 
> > I have replaced it but not sure about the change, please have a look.
> 
> It'll use either CVaaS (if set) or the first server of the first onprem cluster. Maybe something like this then?
> 
> ```diff
>     description: |-
>       Bootstrap URL configured in DHCP to use for inband ZTP.
>       If not set then the bootstrap value will be set to either `https://www.{CV_REGION_TO_SERVER_MAP[next(iter(self.inputs.cv_settings.cvaas.clusters)).region]}/ztp/bootstrap` or
>       `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
>       Otherwise no value will be configured.
> ```

Alternative suggestion to avoid the `next(iter( ...` in documentation - @ClausHolbechArista 

```diff
    description: |-
      Bootstrap URL configured in DHCP to use for inband ZTP.
      If not set then the bootstrap value will be set to match either to
        * if CVaaS is enabled, the URL is formed using the region of the first configured cluster: `https://www.{URL of cv_settings.cvaas.clusters[0].region}/ztp/bootstrap` 
        * if onprem clusters are configured, the URL is formed using the IP or name of the first configured server in the first configured cluster: `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
      Otherwise no value will be configured.
```

ClausHolbechArista commented:
> > > > Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> > > > ```yaml
> > > > keys:
> > > >   inband_ztp_bootstrap_file:
> > > >     type: str
> > > >     description: |-
> > > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > > >       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
> > > >           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
> > > >       Otherwise no value will be configured.
> > > > ```
> > > 
> > > 
> > > I have replaced it but not sure about the change, please have a look.
> > 
> > 
> > It'll use either CVaaS (if set) or the first server of the first onprem cluster. Maybe something like this then?
> > ```diff
> >     description: |-
> >       Bootstrap URL configured in DHCP to use for inband ZTP.
> >       If not set then the bootstrap value will be set to either `https://www.{CV_REGION_TO_SERVER_MAP[next(iter(self.inputs.cv_settings.cvaas.clusters)).region]}/ztp/bootstrap` or
> >       `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
> >       Otherwise no value will be configured.
> > ```
> 
> Alternative suggestion to avoid the `next(iter( ...` in documentation - @ClausHolbechArista
> 
> ```diff
>     description: |-
>       Bootstrap URL configured in DHCP to use for inband ZTP.
>       If not set then the bootstrap value will be set to match either to
>         * if CVaaS is enabled, the URL is formed using the region of the first configured cluster: `https://www.{URL of cv_settings.cvaas.clusters[0].region}/ztp/bootstrap` 
>         * if onprem clusters are configured, the URL is formed using the IP or name of the first configured server in the first configured cluster: `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
>       Otherwise no value will be configured.
> ```

IMO both a way too complex for user-facing documentation.

My go:

```yaml
keys:
  inband_ztp_bootstrap_file:
    type: str
    description: |-
      Bootstrap URL configured in DHCP to use for inband ZTP.
      By default the URL will be `https://<first cv server>/ztp/bootstrap` if `cv_settings` are used. Otherwise no value will be configured.
```

gmuloc commented:
> > > > > Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> > > > > ```yaml
> > > > > keys:
> > > > >   inband_ztp_bootstrap_file:
> > > > >     type: str
> > > > >     description: |-
> > > > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > > > >       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
> > > > >           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
> > > > >       Otherwise no value will be configured.
> > > > > ```
> > > > 
> > > > 
> > > > I have replaced it but not sure about the change, please have a look.
> > > 
> > > 
> > > It'll use either CVaaS (if set) or the first server of the first onprem cluster. Maybe something like this then?
> > > ```diff
> > >     description: |-
> > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > >       If not set then the bootstrap value will be set to either `https://www.{CV_REGION_TO_SERVER_MAP[next(iter(self.inputs.cv_settings.cvaas.clusters)).region]}/ztp/bootstrap` or
> > >       `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
> > >       Otherwise no value will be configured.
> > > ```
> > 
> > 
> > Alternative suggestion to avoid the `next(iter( ...` in documentation - @ClausHolbechArista
> > ```diff
> >     description: |-
> >       Bootstrap URL configured in DHCP to use for inband ZTP.
> >       If not set then the bootstrap value will be set to match either to
> >         * if CVaaS is enabled, the URL is formed using the region of the first configured cluster: `https://www.{URL of cv_settings.cvaas.clusters[0].region}/ztp/bootstrap` 
> >         * if onprem clusters are configured, the URL is formed using the IP or name of the first configured server in the first configured cluster: `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
> >       Otherwise no value will be configured.
> > ```
> 
> IMO both a way too complex for user-facing documentation.
> 
> My go:
> 
> ```yaml
> keys:
>   inband_ztp_bootstrap_file:
>     type: str
>     description: |-
>       Bootstrap URL configured in DHCP to use for inband ZTP.
>       By default the URL will be `https://<first cv server>/ztp/bootstrap` if `cv_settings` are used. Otherwise no value will be configured.
> ```

Thanks

@Vibhu-gslab go with this :)

gmuloc committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab committed:
Fixing description

Vibhu-gslab committed:
Merge branch 'devel' into dep_terminattr

Vibhu-gslab commented:
> > > > > > Description of the `python-avd/pyavd/_eos_designs/schema/schema_fragments/inband_ztp_bootstrap_file.schema.yml` also mentions `cvp_instance_ips[0]` (needs to be replaced with `cvp_server`):
> > > > > > ```yaml
> > > > > > keys:
> > > > > >   inband_ztp_bootstrap_file:
> > > > > >     type: str
> > > > > >     description: |-
> > > > > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > > > > >       If not set and `cvp_instance_ips` is set then the bootstrap value will be set to:
> > > > > >           `https://{cvp_instance_ips[0]}/ztp/bootstrap`
> > > > > >       Otherwise no value will be configured.
> > > > > > ```
> > > > > 
> > > > > 
> > > > > I have replaced it but not sure about the change, please have a look.
> > > > 
> > > > 
> > > > It'll use either CVaaS (if set) or the first server of the first onprem cluster. Maybe something like this then?
> > > > ```diff
> > > >     description: |-
> > > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > > >       If not set then the bootstrap value will be set to either `https://www.{CV_REGION_TO_SERVER_MAP[next(iter(self.inputs.cv_settings.cvaas.clusters)).region]}/ztp/bootstrap` or
> > > >       `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
> > > >       Otherwise no value will be configured.
> > > > ```
> > > 
> > > 
> > > Alternative suggestion to avoid the `next(iter( ...` in documentation - @ClausHolbechArista
> > > ```diff
> > >     description: |-
> > >       Bootstrap URL configured in DHCP to use for inband ZTP.
> > >       If not set then the bootstrap value will be set to match either to
> > >         * if CVaaS is enabled, the URL is formed using the region of the first configured cluster: `https://www.{URL of cv_settings.cvaas.clusters[0].region}/ztp/bootstrap` 
> > >         * if onprem clusters are configured, the URL is formed using the IP or name of the first configured server in the first configured cluster: `https://{cv_settings.onprem_clusters[0].servers[0].name}/ztp/bootstrap`.
> > >       Otherwise no value will be configured.
> > > ```
> > 
> > 
> > IMO both a way too complex for user-facing documentation.
> > My go:
> > ```yaml
> > keys:
> >   inband_ztp_bootstrap_file:
> >     type: str
> >     description: |-
> >       Bootstrap URL configured in DHCP to use for inband ZTP.
> >       By default the URL will be `https://<first cv server>/ztp/bootstrap` if `cv_settings` are used. Otherwise no value will be configured.
> > ```
> 
> Thanks
> 
> @Vibhu-gslab go with this :)

Updated the description :)

gmuloc committed:
Merge branch 'devel' into dep_terminattr

gmuloc commented on docs/porting-guides/6.x.x.md:
```diff
@@ -58,6 +58,10 @@ The following data model keys have been removed from `eos_designs` in AVD 6.0.0.
+| `cvp_ingestauth_key` | - |
+| `cvp_instance_ips` | `cv_settings` |
+| `cvp_token_file` | `cv_settings` |
+| `node_type_keys[].interface_descriptions.overlay_loopback_interface` | `node_type_keys[].interface_descriptions.router_id_loopback_interface`|
```
this one looks unrelated to this PR

Vibhu-gslab replied:
duplicate entry, apologies.. Fixed it.

gmuloc replied:
no worries! :) lot of rebasing going on

gmuloc commented on docs/porting-guides/6.x.x.md:
````diff
@@ -150,6 +158,59 @@ snmp_settings:
+
+    ```eos
+    daemon TerminAttr
+      exec /usr/bin/TerminAttr -cvaddr=192.168.200.12:9910,192.168.200.13:9910 -cvauth=token,/tmp/token -cvvrf=default -disableaaa -smashexcludes=ale,flexCounter,hardware,kni,pulse,strata -taillogs -cvsourceintf=Management0 -smashexcludes=ale,flexCounter,hardware,kni,pulse,strata -ingestexclude=/Sysdb/cell/1/agent,/Sysdb/cell/2/agent
````
why do we have dupicate smash exclude here? is it what we are rendering?

Vibhu-gslab replied:
Fixed it.

gmuloc replied:
thanks!

Vibhu-gslab committed:
Update 6.x.x.md

gmuloc committed:
Merge branch 'devel' into dep_terminattr

gmuloc committed:
Merge branch 'devel' into dep_terminattr
